src/main/java/io/jenkins/docker/client/DockerAPI.java

```diff
@@ -122,21 +137,24 @@ public boolean isSwarm() {
 
     public DockerClient getClient() {
         if (client == null) {
+            final Integer readTimeoutInMillisecondsOrNull = readTimeout > 0 ? Integer.valueOf(readTimeout * 1000) : null;
+            final Integer connectTimeoutInMillisecondsOrNull = connectTimeout > 0 ? Integer.valueOf(connectTimeout * 1000) : null;
             client = DockerClientBuilder.getInstance(
                 new DefaultDockerClientConfig.Builder()
                     .withDockerHost(dockerHost.getUri())
                     .withCustomSslConfig(toSSlConfig(dockerHost.getCredentialsId()))
                 )
                 .withDockerCmdExecFactory(new NettyDockerCmdExecFactory()
-                .withConnectTimeout(connectTimeout))
```

Comment:
Note: I'm pretty sure that this old code was buggy - withConnectTimeout expects to be given milliseconds (or null for "no timeout"), but connectTimeout is in seconds.